CMakeLists.txt

```diff
@@ -0,0 +1,147 @@
+cmake_minimum_required(VERSION 3.8)
+
+set (PROJECT openloco)
+
+project(${PROJECT})
+
+if (CMAKE_BINARY_DIR STREQUAL CMAKE_SOURCE_DIR)
+    message(FATAL_ERROR "Building in-source is not supported! Create a build dir and remove ${CMAKE_SOURCE_DIR}/CMakeCache.txt")
+endif()
+
+INCLUDE(FindPkgConfig)
+
+set(USE_BOOST_FS_DEFAULT OFF)
+if(APPLE)
+	set(USE_BOOST_FS_DEFAULT ON)
+endif()
+
+option(USE_BOOST_FILESYSTEM "Use Boost filesystem instead of C++17" ${USE_BOOST_FS_DEFAULT})
+
+if (APPLE)
+    set(ENV{PKG_CONFIG_PATH} "$ENV{PKG_CONFIG_PATH}:/usr/local/opt/openssl/lib/pkgconfig")
+endif (APPLE)
+
+set(COMMON_COMPILE_OPTIONS "${COMMON_COMPILE_OPTIONS} -fstrict-aliasing -Werror -Wundef -Wmissing-declarations -Winit-self -Warray-bounds -Waddress -Wchar-subscripts -Wenum-compare")
+
+set(OBJ_FORMAT "elf32-i386")
+set(LINKER_SCRIPT "ld_script_i386.xc")
+
+
+# Handle creating the rct2 text and data files on OS X and Linux
+# See details in src/openrct2.c:openrct2_setup_rct2_segment for how the values
+# were derived.
+if (UNIX)
+    set(OLOCO_EXE ${CMAKE_CURRENT_SOURCE_DIR}/loco.exe)
+    set(OLOCO_TEXT ${CMAKE_BINARY_DIR}/openloco_text)
+    set(OLOCO_DATA ${CMAKE_BINARY_DIR}/openloco_data)
+
+    add_custom_command(
+            OUTPUT openloco_text
+            COMMAND dd if="${OLOCO_EXE}" of="${OLOCO_TEXT}" bs=4096 skip=1 count=214
+            DEPENDS ${OLOCO_EXE}
+    )
+    add_custom_command(
+            OUTPUT openloco_data
+            COMMAND dd if="${OLOCO_EXE}" of="${OLOCO_DATA}" bs=4096 skip=215 count=78
```

Comment:
@marijnvdwerf can you double-check these offsets?

Reply:
![image](https://user-images.githubusercontent.com/737603/35188531-f1d27728-fe36-11e7-87a1-eff220010acf.png)

Reply:
text start = 0x1000/4096 = 1
text size: 0xd6000/4096 = 214

rodata start: 0xd7000/4096 = 215
rodata+data size: (0x1f000+0x2f000)/4096 = 78